src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -683,6 +765,140 @@ function InfoBottomSheet({
   )
 }
 
+function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  canSameChainSwapToDeposit,
+  canCrossChainSwapToDeposit,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  canSameChainSwapToDeposit: boolean
+  canCrossChainSwapToDeposit: boolean
+  canAdd: boolean
+  exchanges: ExternalExchangeProvider[]
+}) {
+  const { t } = useTranslation()
+
+  const actions = useMemo(() => {
+    const visibleActions: AddAssetsAction[] = []
+
+    if (canSameChainSwapToDeposit) {
+      visibleActions.push({
+        name: TokenActionName.SwapAndDeposit,
+        details: t(
+          'earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapAndDepositActionDescription',
+          {
+            tokenSymbol: token.symbol,
+            tokenNetwork: NETWORK_NAMES[token.networkId],
+          }
+        ),
+        onPress: () => {
+          AppAnalytics.track(EarnEvents.earn_add_crypto_action_press, {
+            action: TokenActionName.SwapAndDeposit,
+            ...getTokenAnalyticsProps(token),
+          })
+
+          navigate(Screens.SwapScreenWithBack, { toTokenId: token.tokenId })
+        },
+      })
+    }
+
+    if (canCrossChainSwapToDeposit) {
+      visibleActions.push({
+        name: canSameChainSwapToDeposit ? TokenActionName.CrossChainSwap : TokenActionName.Swap,
+        details: canSameChainSwapToDeposit
+          ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainSwapActionDescription', {
+              tokenSymbol: token.symbol,
+              tokenNetwork: NETWORK_NAMES[token.networkId],
+            })
+          : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapActionDescription', {
+              tokenSymbol: token.symbol,
+              tokenNetwork: NETWORK_NAMES[token.networkId],
+            }),
+        onPress: () => {
+          AppAnalytics.track(EarnEvents.earn_add_crypto_action_press, {
+            action: canSameChainSwapToDeposit
+              ? TokenActionName.CrossChainSwap
+              : TokenActionName.Swap,
+            ...getTokenAnalyticsProps(token),
+          })
+
+          navigate(Screens.SwapScreenWithBack, { toTokenId: token.tokenId })
+        },
```

Comment:
With the recent changes, the bottom sheets should be closed or dismissed after navigation.

```suggestion
          navigate(Screens.SwapScreenWithBack, { toTokenId: token.tokenId })
          forwardedRef.current?.close()
        },
```